benchmarks/dbscan/data.hpp

```diff
@@ -0,0 +1,291 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DATA_HPP
+#define DATA_HPP
+
+#include <ArborX_Exception.hpp>
+#include <ArborX_HyperPoint.hpp>
+
+#include <random>
+#include <vector>
+
+using ArborX::ExperimentalHyperGeometry::Point;
+
+template <int DIM>
+std::vector<Point<DIM>> sampleData(std::vector<Point<DIM>> const &data,
+                                   int num_samples)
+{
+  std::vector<Point<DIM>> sampled_data(num_samples);
+
+  std::srand(1337);
+
+  // Knuth algorithm
+  auto const N = (int)data.size();
+  auto const M = num_samples;
+  for (int in = 0, im = 0; in < N && im < M; ++in)
+  {
+    int rn = N - in;
+    int rm = M - im;
+    if (std::rand() % rn < rm)
```

Comment:
`std::rand` is defined in `<cstdlib>` not `<random>` and since you are using the C++11 facility below, prefer `uniform_int_distribution` here